### ISSUE TITLE
more MDAL support TBD in common functions

### ISSUE THREAD

**Issue by jti-lanl** [mar-file-system/marfs#138]
For instance, I just noticed that stat_xattrs() always makes direct calls to lgetxattr(), instead of going through the MDAL.

**Comment by jti-lanl** [comment]
ACCESS() and FACCESSAT() just boil down to direct calls to the POSIX functions.

**Comment by wfvining** [comment]
These functions need to go through the MDAL:
- chown
- mkdir
- rmdir
- access
- readlink
- statvfs
- symlink

The following functions make direct calls to POSIX where they should go through the MDAL
- trash_truncate - common.c
- trunc_xattrs - common.c
- trash_unlink - common.c
- marfs_getfattr - marfs_ops.c
- marfs_create - marfs_ops.c
- init_scatter_tree_internal - common.c
- init_mdfs - common.h

**Comment by wfvining** [comment]
marfs_utime* do not go through the MDAL

**Comment by brettkettering** [comment]
Thanks for documenting, Will.

This is work that needs to be done to complete MDAL and for HB to use MarFS instead of POSIX calls.

HB has his POSIX-based code written. He is testing it to ensure that each role is done properly.

**Comment by jti-lanl** [comment]
For pftool, at least, we also want to support faccessat(..., AT_SYMLINK_NOFOLLOW).  I added marfs_faccessat(), to avoid just bashing marfs_access().  [Currently doesn't go through MDAL.]  We should expand the MDAL, adding faccessat().  It appears to be missing access() as well.

**Comment by wfvining** [comment]
I have added access to the MDAL, I just haven't merged it yet. I'll add faccessat as well. Should the call to faccessat in marfs_access also specify the AT_EACCESS flag to test against the effective uid rather than the real uid (which will always be root)?

**Comment by jti-lanl** [comment]
Actually, I've added a marfs_faccessat(), which just gets its flags from the caller.  Changed pftool to use that instead of access(), because it was trying to chase symlinks.  (I added an faccessat() to Path subclasses, instead of subverting access(), in case we actually do want access() someday.)

So, both marfs_faccessat() and marfs_access() are in need of MDAL-ification, from my perspective.  I'll push pretty soon, after just a little more testing.
